### DISCUSSION TITLE
Prisma Connections

### DISCUSSION THREAD

**Discussion by MarwanRadwan7** [prisma/prisma#21672] [Q&A] [answered]
### Question

Hello, I was reviewing a repo for a backend application using typescript, prisma and postgresql and I found this line `public users = new PrismaClient().user;` and similar in each class like categories, books, etc... I just wanted to ask: when we make this instance in every class means we are creating a new connection to the database. Does this make an excessive use of the database pools ? Or how does this work ! How should I use it effeciently. 🙌

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
  public users = new PrismaClient().user;

```

- OS: linux
- Database: PostgreSQL
- Node.js version: v18.18.0

-prisma                  : 4.16.2
-@prisma/client          : 4.16.2

**Comment by janpio** [comment, marked as answer]
Yes, this creates a new instance of PrismaClient each time with its own connection pool. Probably not a good idea, use some kind of singleton instead so it uses the same instance for all of these.
